WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCase.kt

```diff
@@ -79,4 +84,12 @@ class GetBackupDownloadStatusUseCase @Inject constructor(
             false
         }
     }
+
+    private suspend fun FlowCollector<BackupDownloadRequestState>.exceedsRetryAttempts(retryAttempts: Int): Boolean {
+        return if (retryAttempts >= MAX_RETRY) {
+            AppLog.d(T.JETPACK_BACKUP, "$tag: Exceeded $MAX_RETRY retries while fetching status")
+            emit(RemoteRequestFailure)
```

Comment:
💡 It feels a bit unexpected that a method called `exceedsRetryAttempts` emits a value. We might want to consider renaming it and moving even the delay here. Something like this:

```
private suspend fun FlowCollector<BackupDownloadRequestState>.handleError(retryAttempts: Int): Boolean {
        return if (retryAttempts >= MAX_RETRY) {
            AppLog.d(T.JETPACK_BACKUP, "$tag: Exceeded $MAX_RETRY retries while fetching status")
            emit(RemoteRequestFailure)
            true
        } else {
            delay(DELAY_MILLIS * (max(1, DELAY_FACTOR * retryAttempts)))
            false
        }
    }
```

Wdyt?

Reply:
I originally named the fun `handleError`, but felt the name didn't invoke a question that matched with a boolean return value. `exceedsRetryAttempts` feels like a question that would return a true/false value. The reverse can be said with emitting a value with the name `handleError` vs. `exceedsRetryAttempts`. So I followed the `isNetworkAvailable` pattern here .. ask question, emit if needed, return boolean.

I agree with moving the backoff logic here.

Reply:
Renamed in 83916f7

Reply:
> I originally named the fun handleError, but felt the name didn't invoke a question that matched with a boolean return value

I see, but I think it's a common pattern on Android to return "true" when an event was handled and "false" when it wasn't handled. If we wanted to follow the Android pattern even more we could call it "OnErrorReceived" but I don't think it's necessary, since this is not a system/user event.

Reply:
@malinajirka - That's fine, I had already renamed it, just wanted you to know why I had the original name.